### PR TITLE
Systhreads library: merge all C code in a single file st_stubs.c

### PR DESCRIPTION
**Pull request by xavierleroy**
The C stub code in otherlibs/systhreads used to be carefully structured (shall I say functorized?) into OS-independent code in `st_stubs.c`, relying on a thin abstraction of the OS thread API, and two implementations of this API, one for POSIX threads and the other for Win32.

Since version 5, OCaml is married to the POSIX threads API, using the winpthreads library to provide this API under Windows.  All this parameterization by the OS is no longer necessary.

This PR shows what happens if all the C code for the systhreads library is grouped together in a single `st_stubs.c` file.  It is a second step towards #13348 .

### CODE REVIEW

**Comment by gasche** [comment]
I'm nervous about the interaction between this approach and #13416, which removes winpthreads and adds a conditional on the operating system again. Maybe it is possible to redo #13416 on top of your PR, but maybe it is easier to see what a final design would look like by working from the result of #13416.

(This is, of course, assuming that we find #13416 desirable and decide to merge it. I'm not a Windows person, so I cannot really judge, but I notice that about 100% of our Windows people are enthusiastic about it.)

File: otherlibs/systhreads/st_stubs.c [hunk 1]
```diff
@@ -29,7 +29,24 @@
 #define DLL_EXPORT
 #endif
```
Comment (MisterDA) [inline otherlibs/systhreads/st_stubs.c:31]:
```suggestion
#include "caml/config.h"
```

for the configure-defined macros

**Review by MisterDA — COMMENTED** [review]
Being the author of #13416, I think that if we remove winpthreads it's nicer to decouple the abstractions over pthreads and the WinAPI, and the common code. If we remove winpthreads, I also think that `st_posix.h` and `st_pthreads.h` could be merged (I've suggested it in #13352).
I can always rebase #13416 but oof. There's already new work for me since #13429! I appreciate if I could do only one rebase ;)

**Comment by xavierleroy** [comment]
Ah, I missed #13416 .  Not sure I agree about ending the pthread marriage.  Until this is decided, closing this PR.

**Pull request closed without merging**